kubemarine/procedures/restore.py

```diff
@@ -135,19 +136,25 @@ def restore_thirdparties(cluster):
 
     install.system_prepare_thirdparties(cluster)
 
-
+    
 def import_nodes(cluster):
     for node in cluster.nodes['all'].get_ordered_members_list(provide_node_configs=True):
-        node['connection'].put(os.path.join(cluster.context['backup_tmpdir'], 'nodes_data', '%s.tar.gz' % node['name']),
-                               '/tmp/kubemarine-backup.tar.gz')
-        cluster.log.debug('Backup \'%s\' uploaded' % node['name'])
-
-    cluster.log.debug('Unpacking backup...')
-    result = cluster.nodes['all'].sudo(
-        'chattr -i /etc/resolv.conf; sudo tar xzvf /tmp/kubemarine-backup.tar.gz -C / --overwrite && sudo chattr +i /etc/resolv.conf')
-    cluster.log.debug(result)
-
-
+        node_backup_file = os.path.join(cluster.context['backup_tmpdir'], 'nodes_data', '%s.tar.gz' % node['name'])
+        if os.path.exists(node_backup_file):
+            node['connection'].put(node_backup_file, '/tmp/kubemarine-backup.tar.gz')
+            cluster.log.debug('Backup \'%s\' uploaded' % node['name'])
+
+            if not cluster.procedure_inventory.get('backup_plan', {}).get('nodes', {}).get('resolv.conf'):
```

Comment:
procedure_inventory for `restore` procedure cannot have `backup_plan` property. See https://github.com/Netcracker/KubeMarine/blob/main/documentation/Maintenance.md#restore-procedure

Reply:
In that case, we can remove the following lines:

```
if not cluster.procedure_inventory.get('backup_plan', {}).get('nodes', {}).get('resolv.conf'):
                restore_command = 'sudo tar xzvf /tmp/kubemarine-backup.tar.gz --exclude=/etc/resolv.conf -C / --overwrite'
            else:
                restore_command = 'sudo tar xzvf /tmp/kubemarine-backup.tar.gz -C / --overwrite'
```

And replace them with just
`restore_command = 'sudo tar xzvf /tmp/kubemarine-backup.tar.gz -C / --overwrite'`



The entire code would look something like this, would this be OK?

```
def import_nodes(cluster):
    for node in cluster.nodes['all'].get_ordered_members_list(provide_node_configs=True):
        node_backup_file = os.path.join(cluster.context['backup_tmpdir'], 'nodes_data', '%s.tar.gz' % node['name'])
        if os.path.exists(node_backup_file):
            node['connection'].put(node_backup_file, '/tmp/kubemarine-backup.tar.gz')
            cluster.log.debug('Backup \'%s\' uploaded' % node['name'])

            if 'backup_plan' in cluster.procedure_inventory.get('restore', {}):
                raise ValueError('The procedure_inventory for restore procedure cannot have backup_plan property')

            restore_command = 'sudo tar xzvf /tmp/kubemarine-backup.tar.gz -C / --overwrite'
            if 'nodes' in cluster.procedure_inventory.get('restore', {}):
                nodes = cluster.procedure_inventory['restore']['nodes']
                if nodes.get('resolv.conf', False):
                    restore_command = 'sudo tar xzvf /tmp/kubemarine-backup.tar.gz --exclude=/etc/resolv.conf -C / --overwrite'

            result = node['connection'].sudo(restore_command)
            cluster.log.debug(result)
        else:
            raise FileNotFoundError('Backup file for node \'%s\' not found' % node['name'])
```

Reply:
* There can never be cluster.procedure_inventory['restore']. All possible parameters are present in the mentioned documentation, or in JSON schema (see its mention in the same documentation).
* There previously was a `chattr -i /etc/resolv.conf` but it is removed in your last solution. Was it done intentionally?

Reply:
Thank you, much better this time.